cypress/integration/Regression/CreateModels/ApiCallbacks.spec.js

```diff
@@ -0,0 +1,91 @@
+      models.CreateNewModel('z-ApiCallbacks')
+    })
+  })
+
+  context('Create Entities', () => {
+    it('Should create custom trained entities that will be used by the API Callbacks', () => {
+      entities.CreateNewEntity({ type: 'Custom Trained', name: '1stArg'})
+      entities.CreateNewEntity({ type: 'Custom Trained', name: '2ndArg'})
+      // entities.CreateNewEntity({ type: 'Custom Trained', name: 'entityError'})
+      // entities.CreateNewEntity({ type: 'Custom Trained', name: 'logicError'})
+      // entities.CreateNewEntity({ type: 'Custom Trained', name: 'renderError'})
+    })
+  })
+
+  context('Create Actions', () => {
+    it('Should create BadCard Action', () => {
+      actions.CreateNewActionThenVerifyInGrid({ 
+        responseNameData: 'BadCard',
+        type: 'API',
+        validateApiResponse: 'BadCardrender(result, memoryManager)'
+      })
+    })
+
+    it('Should create ExceptionAPI Action', () => {
+      actions.CreateNewActionThenVerifyInGrid({ 
+        responseNameData: 'ExceptionAPI', 
+        type: 'API',
+        validateApiResponse: 'ExceptionAPIlogic(memoryManager)render(result, memoryManager)' 
+      })
+    })
+
+    it('Should create LogicWithNoArgs Action', () => {
+      actions.CreateNewActionThenVerifyInGrid({ 
+        responseNameData: 'LogicWithNoArgs', 
+        type: 'API',
+        validateApiResponse: 'LogicWithNoArgslogic(memoryManager)'
+      })
+    })
+    
+    it('Should create LogicWithArgs Action', () => {
+      actions.CreateNewActionThenVerifyInGrid({ 
+        responseNameData: 'LogicWithArgs', 
+        type: 'API',
+        logicArgs: ['$1stArg{enter}', '$2ndArg{enter}'],
+        validateApiResponse: 'LogicWithArgslogic(memoryManager, firstArg, secondArg)firstArg:"$1stArg"secondArg:"$2ndArg"'
+      })
+    })
+
+    it('Should create Malformed Action', () => {
+      actions.CreateNewActionThenVerifyInGrid({ 
+        responseNameData: 'Malformed', 
+        type: 'API',
+        validateApiResponse: 'Malformedlogic(memoryManager)'
+      })
+    })
+    
+    it('Should create RenderTheArgs Action', () => {
+      actions.CreateNewActionThenVerifyInGrid({ 
+        responseNameData: 'RenderTheArgs',
+        type: 'API',
+        logicArgs: ['$1stArg{enter}', '$2ndArg{enter}', '333', '4444', 'five', 'six', 'seven'],                                          
+        renderArgs: ['$1stArg{enter}', '$2ndArg{enter}', 'three', 'four', '55555', '666666', '7777777'],
+        validateApiResponse: 'RenderTheArgslogic(memoryManager, firstArg, secondArg, thirdArg, fourthArg, fifthArg, sixthArg, seventhArg)firstArg:"$1stArg"secondArg:"$2ndArg"thirdArg:"333"fourthArg:"4444"fifthArg:"five"sixthArg:"six"seventhArg:"seven"render(result, memoryManager, firstArg, secondArg, thirdArg, fourthArg, fifthArg, sixthArg, seventhArg)firstArg:"$1stArg"secondArg:"$2ndArg"thirdArg:"three"fourthArg:"four"fifthArg:"55555"sixthArg:"666666"seventhArg:"7777777"'
```

Comment:
I don't think you would want to validate the response this way.  The display/format has changed and might to change again which would break tests.

The string is also not an expected format for people writing tests.  This relies on the joining of text without newlines and other internal behavior. 

Many disagreements with the function but I guess we'll put that aside.  If you want to do this type of validation I try to find a way to construct this string from the other input (response and logic /render args) so is always correct and people don't have to manually type it. Could easily be a typo or wrong character.

Reply:
Even if I construct from the arguments, if the format changes on the UI I'll still have to change the test code.

I don't have to worry about typos since I grab this string from the logs...I run the test with this value undefined, grab the string from the logs and past it into the code...of course I look it over for a sanity check before I do to ensure the output makes sense. So if it does fail because the UI changes, that change will appear in the log file...I just have to copy and paste it into the test code to fix the test error.

Also when output like this changes, it is nice for me to discover that and have to change the code. This way I am one more pair of eyes on the change to confirm it makes sense to someone besides the developers who put it there.